optuna_dashboard/ts/components/GraphHistory.tsx

```diff
@@ -247,62 +253,73 @@ const plotHistory = (
-    const yForLinePlot: number[] = []
-    let currentBest: number | null = null
-    for (let i = 0; i < trials.length; i++) {
-      const t = trials[i]
-      if (currentBest === null) {
-        currentBest = t.values![objectiveId] as number
-        xForLinePlot.push(getAxisX(t))
-        yForLinePlot.push(t.values![objectiveId] as number)
-      } else if (
-        directions[objectiveId] === "maximize" &&
-        t.values![objectiveId] > currentBest
-      ) {
-        const p = trials[i - 1]
-        if (!xForLinePlot.includes(getAxisX(p))) {
-          xForLinePlot.push(getAxisX(p))
-          yForLinePlot.push(currentBest)
+    const objectiveId = target.getObjectiveId()
+    if (objectiveId !== null) {
+      const xForLinePlot: (number | Date)[] = []
+      const yForLinePlot: number[] = []
+      let currentBest: number | null = null
+      for (let i = 0; i < h.trials.length; i++) {
+        const t = h.trials[i]
+        const value = target.getTargetValue(t) as number
+        if (value === null) {
+          continue
+        } else if (currentBest === null) {
+          currentBest = value
+          xForLinePlot.push(getAxisX(t))
+          yForLinePlot.push(value)
+        } else if (
+          h.directions[objectiveId] === "maximize" &&
+          value > currentBest
+        ) {
+          const p = h.trials[i - 1]
+          if (!xForLinePlot.includes(getAxisX(p))) {
+            xForLinePlot.push(getAxisX(p))
+            yForLinePlot.push(currentBest)
+          }
+          currentBest = value
+          xForLinePlot.push(getAxisX(t))
+          yForLinePlot.push(value)
+        } else if (
+          h.directions[objectiveId] === "minimize" &&
+          value < currentBest
+        ) {
+          const p = h.trials[i - 1]
+          if (!xForLinePlot.includes(getAxisX(p))) {
+            xForLinePlot.push(getAxisX(p))
+            yForLinePlot.push(currentBest)
+          }
+          currentBest = value
+          xForLinePlot.push(getAxisX(t))
+          yForLinePlot.push(value)
         }
-        currentBest = t.values![objectiveId] as number
-        xForLinePlot.push(getAxisX(t))
-        yForLinePlot.push(t.values![objectiveId] as number)
-      } else if (
-        directions[objectiveId] === "minimize" &&
-        t.values![objectiveId] < currentBest
-      ) {
-        const p = trials[i - 1]
-        if (!xForLinePlot.includes(getAxisX(p))) {
-          xForLinePlot.push(getAxisX(p))
-          yForLinePlot.push(currentBest)
-        }
-        currentBest = t.values![objectiveId] as number
-        xForLinePlot.push(getAxisX(t))
-        yForLinePlot.push(t.values![objectiveId] as number)
       }
+      if (h.trials.length !== 0) {
+        xForLinePlot.push(getAxisX(h.trials[h.trials.length - 1]))
+        yForLinePlot.push(yForLinePlot[yForLinePlot.length - 1])
+      }
+      plotData.push({
+        x: xForLinePlot,
+        y: yForLinePlot,
+        name: `Best Value of ${h.study_name}`,
```

Comment:
📝 A minor note but we now always append the name of the study, even when there's only a single study.